src/api-v1.js

```diff
@@ -99,7 +99,15 @@ api.declare({
       // instead of a more complicated structure
 
       if (x === 0 && (result.status === 'absent' || result.status === 'error')) {
-        let validUrl = await validateUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL, this.monitor);
+        let validUrl;
+        try {
+          validUrl = await validateUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL, this.monitor);
```

Comment:
Okay, I guess it's no worse than before... I just don't get why we don't look up the cache and the backing store before we start look at redirects and other things... it seems expensive...